libs/SalesforceSDK/src/com/salesforce/androidsdk/app/SalesforceSDKManager.java

```diff
@@ -152,6 +161,19 @@ public class SalesforceSDKManager implements LifecycleObserver {
     protected Context context;
     private LoginOptions loginOptions;
     private final Class<? extends Activity> mainActivityClass;
+
+    /**
+     * Null or an authenticated Activity for private use when developer support
+     * is enabled.
+     */
+    private Activity authenticatedActivityForDeveloperSupport = null;
```

Comment:
This is perhaps the most interesting part of the whole effort.  Requesting permissions requires an activity.  Responding to a notification action, in newer versions of Android, requires a call to `Activity.startActivity` even though we only want to show a dialog on the existing activity - anything else doesn't close the system drawer.  All the developer support dialog management is done here, in a singleton that only has an Application for context.

To integrate this with a smaller code footprint, this Activity variable keeps a reference to the top Activity post-login.  Further on, this is set by a utility method called from an Activity Lifecycle Callback.  That provides an Activity to request permissions.  Also, a new `ShowDeveloperSupportActivity` acts as a disposable trigger from the notification action.

I had a few other ideas to integration the permissions management and notification action, but they started to present API change to apps.  The approach here became the best black box integration in my trials.  Any other ideas?  I like the way this turned out, especially that it's entirely under-the-hood to apps.  I'd be game to evolve it in a different direction also.

Reply:
I should mention, the `context` variable on `SalesforceSDKManager` gets an inspector flag by Android Studio.  The flow of Context for Activity and Application can get some future polish in the modernization epic.